libs/ais-service/src/main/java/uk/gov/di/ipv/core/library/ais/client/AisClient.java

```
@@ -0,0 +1,162 @@
+                    HttpRequest.newBuilder()
+                            .uri(getUri(userId))
+                            .GET()
+                            .header(CONTENT_TYPE, ContentType.APPLICATION_JSON.toString());
+
+            var response = sendHttpRequest(httpRequestBuilder.build());
+            return OBJECT_MAPPER.readValue(response.body(), AccountInterventionStatusDto.class);
+        } catch (JsonProcessingException e) {
+            throw new AisClientException("Failed parse AIS response body", e);
+        }
+    }
+
+    private URI getUri(String userId) throws AisClientException {
+        try {
+            var baseUri =
+                    "%s/ais/%s"
+                            .formatted(
+                                    configService.getParameter(AIS_API_BASE_URL),
+                                    URLEncoder.encode(userId, StandardCharsets.UTF_8));
+            var uriBuilder = new URIBuilder(baseUri);
+            return uriBuilder.build();
+        } catch (URISyntaxException e) {
+            throw new AisClientException("Failed to construct AIS URL", e);
+        }
+    }
+
+    private HttpResponse<String> sendHttpRequest(HttpRequest aisHttpRequest)
+            throws AisClientException {
+
+        try {
+            return Retry.runTaskWithBackoff(
+                    sleeper,
+                    NUMBER_OF_HTTP_REQUEST_ATTEMPTS,
+                    RETRY_DELAY_MILLIS,
+                    () -> {
+                        LOGGER.info(LogHelper.buildLogMessage("Sending HTTP request to AIS"));
+                        try {
+                            var response =
+                                    httpClient.send(
+                                            aisHttpRequest, HttpResponse.BodyHandlers.ofString());
+                            checkResponseStatusCode(response);
+                            return response;
+                        } catch (IOException e) {
```

The EvcsClient code that is similar to this treats IOExceptions as non-retriable, and only retries on 401 (unauthorised) and 429 (too many requests).
I would have though IOExceptions are likely to be transitory things and worth retrying whereas 401 seems unlikely to be fixed by retrying.

---

@blakeyp If you agree with this, is it worth raising a tech debt ticket for the EvcsClient code

---

@DanCorderIPV we have some special handling for IOExceptions in the base http client where we retry on certain types of IOException and fail hard on anything else (due to the recent incidents where lambda instances appear to lose network connectivity and all requests fail with IOException)
https://github.com/govuk-one-login/ipv-core-back/blob/c8c8a91da1fb2b6a64f2f0b176503846729b38da/libs/common-services/src/main/java/uk/gov/di/ipv/core/library/tracing/TracingHttpClient.java#L56